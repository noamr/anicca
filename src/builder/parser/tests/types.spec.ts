/// </ reference "jest" />
import nearley from 'nearley'
const typeGrammar = require('../types.ne')

const parse = (str: string) => new nearley.Parser(typeGrammar).feed(str).finish()[0]

describe('type grammar', () => {
    it('singular', () => {
        expect(parse('u32')).toMatchSnapshot()
        expect(parse('f64')).toMatchSnapshot()
        expect(parse('string')).toMatchSnapshot()
    })

    it('multi', () => {
        expect(parse('[u32, string]')).toMatchSnapshot()
        expect(parse('[string, bool]')).toMatchSnapshot()
        expect(parse('{string: u32}')).toMatchSnapshot()
    })

    it('compound', () => {
        expect(parse('[u32, [f32, [string, bool]]]')).toMatchSnapshot()
        expect(parse('[string, {bool: {string: [i8, bool]}}]')).toMatchSnapshot()

    })
})
