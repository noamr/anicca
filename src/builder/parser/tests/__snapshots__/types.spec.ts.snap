// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type grammar compound 1`] = `
Object {
  "tuple": Array [
    "u32",
    Object {
      "tuple": Array [
        "f32",
        Object {
          "tuple": Array [
            "string",
            "bool",
          ],
        },
      ],
    },
  ],
}
`;

exports[`type grammar compound 2`] = `
Object {
  "tuple": Array [
    "string",
    Object {
      "dictionary": Array [
        "bool",
        Object {
          "dictionary": Array [
            "string",
            Object {
              "tuple": Array [
                "i8",
                "bool",
              ],
            },
          ],
        },
      ],
    },
  ],
}
`;

exports[`type grammar multi 1`] = `
Object {
  "tuple": Array [
    "u32",
    "string",
  ],
}
`;

exports[`type grammar multi 2`] = `
Object {
  "tuple": Array [
    "string",
    "bool",
  ],
}
`;

exports[`type grammar multi 3`] = `
Object {
  "dictionary": Array [
    "string",
    "u32",
  ],
}
`;

exports[`type grammar singular 1`] = `"u32"`;

exports[`type grammar singular 2`] = `"f64"`;

exports[`type grammar singular 3`] = `"string"`;
